alpha/lib/model/kuserPeer.php

```diff
@@ -707,6 +707,24 @@ public static function sendNewUserMail(kuser $user, $existingUser)
 			);
 		}
 	}
+	
+	protected static function getAuthenticationType($user, $partner)
+	{
+		/* @var $user kuser*/
+		$authType = $partner->getAuthenticationType();
+		if ($authType == PartnerAuthenticationType::SSO)
+		{
+			if ($user->getIsSsoExcluded())
```

Comment:
you can just ask if ($user->getIsSsoExcluded() && !$partner->getUseTwoFactorAuthentication())
then return password only
otherwise you can just return the auth type of the partner

Reply:
what you are suggesting is missing some use cases